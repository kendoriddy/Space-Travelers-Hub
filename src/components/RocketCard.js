import React from 'react';
import { useSelector } from 'react-redux';

const RocketCard = () => {
  const rockets = useSelector((state) => state.rockets);

  const renderRocket = rockets.map((rocket) => (
    <div key={rocket.id} className="main">
      <div className="rocketImage">
        <img src={rocket.flickr_images} alt={rocket.rocket_name} />
      </div>
      <div className="rocketDescription">
        <h1>{rocket.rocket_name}</h1>
        <p>
          {rocket.description}
        </p>
        <button type="button">Reserved</button>
      </div>
    </div>
  ));

  return (
    <>{ renderRocket }</>
  );
};

export default RocketCard;
