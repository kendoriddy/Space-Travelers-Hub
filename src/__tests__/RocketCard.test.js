import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketCard from '../components/RocketCard';

it('renders as expected', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <RocketCard key={0} id={0} reserved={false} name="rocket.rocket_name" description="rocket.description" img={[]} bookingHandler={() => {}} />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
