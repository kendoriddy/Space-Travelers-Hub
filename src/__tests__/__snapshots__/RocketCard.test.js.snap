// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="container"
>
  <div
    className="rocketBody"
  >
    <div
      className="rocketImage"
    >
      <img
        alt=""
        src={Array []}
      />
    </div>
    <div
      className="rocketDescription"
    >
      <h1>
        rocket.rocket_name
      </h1>
      <p>
        rocket.description
      </p>
      <button
        className="btn btn-reserve width-20vw btn btn-primary"
        disabled={false}
        id={0}
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
